msbuild/Xamarin.MacDev.Tasks.Core/Tasks/ComputeBundleLocationTaskBase.cs

```diff
@@ -0,0 +1,333 @@
+				return;
+			}
+
+			Log.LogWarning (MSBStrings.E7088 /* The 'PublishFolderType' metadata value '{0}' on the item '{1}' is not recognized. The file will not be copied to the app bundle. */, publishFolderType, item.ItemSpec);
+		}
+
+		// 'item' is not supposed to have a PublishFolderType set
+		PublishFolderType ComputePublishFolderType (IList<ITaskItem> items, ITaskItem item)
+		{
+			var filename = item.ItemSpec;
+			var targetPath = item.GetMetadata ("TargetPath");
+			if (!string.IsNullOrEmpty (targetPath))
+				filename = Path.Combine (Path.GetDirectoryName (filename), Path.GetFileName (targetPath));
+
+			// Check if the item came from @(BundleResource), @(Content) or @(EmbeddedResource)
+			if (resourceFilesSet.Contains (Path.GetFullPath (item.ItemSpec)))
+				return PublishFolderType.Resource;
+
+			// Assemblies and their related files
+			if (filename.EndsWith (".dll", StringComparison.OrdinalIgnoreCase)) {
+				return PublishFolderType.Assembly;
+			} else if (filename.EndsWith (".exe", StringComparison.OrdinalIgnoreCase)) {
+				return PublishFolderType.Assembly;
+			} else if (filename.EndsWith (".pdb", StringComparison.OrdinalIgnoreCase)) {
+				return PublishFolderType.Assembly;
+			} else if (filename.EndsWith (".dll.mdb", StringComparison.OrdinalIgnoreCase) || filename.EndsWith (".exe.mdb", StringComparison.OrdinalIgnoreCase)) {
+				return PublishFolderType.Assembly;
+			} else if (filename.EndsWith (".config", StringComparison.OrdinalIgnoreCase)) {
+				return PublishFolderType.Assembly;
+			}
+
+			// Binding resource package (*.resources / *.resources.zip)
+			if (IsBindingResourcePackage (filename, out var type))
+				return type;
+
+			// Native (xc)frameworks.
+			// We do this after checking for binding resource packages, because those might contain frameworks.
+			if (TryGetFrameworkDirectory (filename, out _))
+				return PublishFolderType.AppleFramework;
+
+			// resources (png, jpg, ...?)
+			if (filename.EndsWith (".jpg", StringComparison.OrdinalIgnoreCase)) {
+				return PublishFolderType.Resource;
+			} else if (filename.EndsWith (".png", StringComparison.OrdinalIgnoreCase)) {
+				return PublishFolderType.Resource;
+			}
+
+			// *.framework.zip, *.xcframework.zip
+			if (filename.EndsWith (".framework.zip", StringComparison.OrdinalIgnoreCase)) {
+				return PublishFolderType.CompressedAppleFramework;
+			} else if (filename.EndsWith (".xcframework.zip", StringComparison.OrdinalIgnoreCase)) {
+				return PublishFolderType.CompressedAppleFramework;
+			}
+
+			// *.a and *.dylib
+			if (filename.EndsWith (".a", StringComparison.OrdinalIgnoreCase)) {
+				return PublishFolderType.StaticLibrary;
+			} else if (filename.EndsWith (".dylib", StringComparison.OrdinalIgnoreCase)) {
+				return PublishFolderType.DynamicLibrary;
+			}
```

Comment:
Same.

Reply:
This returns different values, so a single if and || won't work.